src/main/java/graphql/schema/DataFetchingEnvironment.java

```diff
@@ -100,4 +108,18 @@
      * @return the {@link DataFetchingFieldSelectionSet} for the current operation
      */
     DataFetchingFieldSelectionSet getSelectionSet();
+
+
+    /**
+     * This allows you to transform this DataFetchingEnvironment object into another one.
+     *
+     * @param action the action to which a {@link DataFetchingEnvironmentBuilder} is passed
+     *
+     * @return a new transformed DataFetchingEnvironment
+     */
+    default DataFetchingEnvironment transform(Consumer<DataFetchingEnvironmentBuilder> action) {
```

Comment:
Where would you put it? On the DFEImpl instead?  

I can imagine DataFetchers that wrap other data fetchers might want to use this to "tweak" the environment before calling the wrapped DF.

But its just sugar on `Builder.newDataFetchingEvironment(currentEv).build() so I can remove it if you want.  I am not wedded to it in this case